build/cmake/functions.cmake

```diff
@@ -422,42 +432,35 @@ macro(wx_add_library name)
             set(runtime_dir "bin")
         endif()
         wx_install(TARGETS ${name}
+            EXPORT ${name}Targets
             LIBRARY DESTINATION "lib/${GEN_EXPR_DIR_FIX}${wxPLATFORM_LIB_DIR}"
             ARCHIVE DESTINATION "lib/${GEN_EXPR_DIR_FIX}${wxPLATFORM_LIB_DIR}"
             RUNTIME DESTINATION "${runtime_dir}/${GEN_EXPR_DIR_FIX}${wxPLATFORM_LIB_DIR}"
             BUNDLE DESTINATION Applications/wxWidgets
             )
+        wx_install(EXPORT ${name}Targets NAMESPACE wx:: DESTINATION "lib/${wxPLATFORM_LIB_DIR}/cmake")
+        wx_target_enable_precomp(${name} "${wxSOURCE_DIR}/include/wx/wxprec.h")
     endif()
 endmacro()
 
-# Enable cotire for target, use optional second argument for prec. header
-macro(wx_target_enable_precomp target_name)
-    target_compile_definitions(${target_name} PRIVATE WX_PRECOMP)
-    if(${ARGC} GREATER 1 AND NOT ${ARGV1} STREQUAL "")
-        set_target_properties(${target_name} PROPERTIES
-            COTIRE_CXX_PREFIX_HEADER_INIT ${ARGV1})
-    endif()
-    set_target_properties(${target_name} PROPERTIES COTIRE_ADD_UNITY_BUILD FALSE)
-    cotire(${target_name})
-endmacro()
-
-# Enable precompiled headers for applications
-macro(wx_app_enable_precomp target_name)
-    if(wxBUILD_PRECOMP)
-        wx_target_enable_precomp(${target_name} "${wxSOURCE_DIR}/include/wx/wxprec.h")
-    elseif(MSVC)
-        target_compile_definitions(${target_name} PRIVATE NOPCH)
-    endif()
-endmacro()
-
-# Enable precompiled headers for wx libraries
-macro(wx_finalize_lib target_name)
+# Enable precompiled headers for target
+macro(wx_target_enable_precomp target_name prec_header)
     if(wxBUILD_PRECOMP)
-        if(TARGET ${target_name})
-            wx_target_enable_precomp(${target_name} "${wxSOURCE_DIR}/include/wx/wxprec.h")
+        target_compile_definitions(${target_name} PRIVATE WX_PRECOMP)
+        if(CMAKE_VERSION VERSION_LESS "3.16" OR wxBUILD_PRECOMP STREQUAL "COTIRE")
+            set_target_properties(${target_name} PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT ${prec_header})
+            set_target_properties(${target_name} PROPERTIES COTIRE_ADD_UNITY_BUILD FALSE)
+            cotire(${target_name})
+        else()
+            get_target_property(target_source_files ${target_name} SOURCES)
+            list(FILTER target_source_files INCLUDE REGEX ".*(.cpp|.cxx)$")
+            list(LENGTH target_source_files target_source_count)
+            if(target_source_count GREATER_EQUAL 2)
+                target_precompile_headers(${target_name} PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${prec_header}>")
+            endif()
```

Comment:
`else`?

Reply:
Building a PCH for a project with only one source files adds more overhead than is gained by the PCH. Cotire has a minimum of two files too.
I'll add a comment to clarify.